api/base/serializers.py

```diff
@@ -44,6 +44,48 @@ def format_relationship_links(related_link=None, self_link=None, rel_meta=None,
     return ret
 
 
+class HideIfRetraction(ser.Field):
+    """
+    If node is retracted, this field will return None.
+    """
+    def __init__(self, field, **kwargs):
+        super(HideIfRetraction, self).__init__(**kwargs)
+        self.field = field
+        self.source = field.source
+        self.required = field.required
+        self.read_only = field.read_only
+
+    def get_attribute(self, instance):
+        if instance.is_retracted:
+            return None
+        return self.field.get_attribute(instance)
+
+    def bind(self, field_name, parent):
+        super(HideIfRetraction, self).bind(field_name, parent)
+        self.field.bind(field_name, self)
+
+    def to_internal_value(self, data):
+        return self.field.to_internal_value(data)
+
+    def to_representation(self, value):
+        if getattr(self.field.root, 'child', None):
```

Comment:
If I set these attributes on __init__ there is an issue with SerializerMethodFields nested within HideIfRetraction.